File: exercism.io/javascript/saddle-points/saddle-points.js
//
// This is only a SKELETON file for the 'Saddle Points' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(matrix) {
    this.matrix = matrix.split("\n")
    for(let i = 0; i < this.matrix.length; i++){
      this.matrix[i] = this.matrix[i].trim().split(" ")
      for(let j = 0; j < this.matrix[i].length; j++){
        this.matrix[i][j] = parseInt(this.matrix[i][j])
      }
    }
  }

  get rows() {
    return this.matrix
  }

  get columns() {
    let result = []
    for(let i = 0; i < this.matrix[0].length; i++){
      let minimums = []
      for(let j = 0; j < this.matrix.length; j++){
        minimums.push(this.matrix[j][i])
      }
      result.push(minimums)
    }
    return result
  }

  get saddlePoints() {
    let result = []
    for(let i = 0; i < this.matrix.length; i++){
      for(let j = 0; j < this.matrix[i].length; j++){
        if(this.matrix[i][j] == Math.max(...this.rows[i]) && this.matrix[i][j] == Math.min(...this.columns[j])){ result.push([i, j]) }
      }
    }
    return result
  }
}

File: exercism.io/python/saddle-points/saddle_points.py
def saddle_points(matrix):
    
    if len(matrix) == 0:
        return [{}]

    lenRow = len(matrix[-1])
    for i in range(0, len(matrix)):
        if len(matrix[i]) != lenRow:
            raise ValueError(".+")

    rows = [max(nums) for nums in matrix]
    columns = [min(nums) for nums in zip(*matrix)]

    result = []
    for i in range(0, len(matrix)):
        for j in range(0, len(matrix[i])):
            if matrix[i][j] == columns[j] and matrix[i][j] == rows[i]:
                result.append({"row": i+1, "column": j+1})

    return [{}] if len(result) == 0 else result
